controllers/csiscaleoperator_controller.go

```diff
@@ -577,45 +581,45 @@ func (r *CSIScaleOperatorReconciler) restartControllerPod(logger logr.Logger, in
 
 	logger.Info("restarting Controller Pod")
 	controllerPod := &corev1.Pod{}
-	controllerStatefulset, err := r.getControllerStatefulSet(instance)
+	controllerDeployment, err := r.getControllerDeployment(instance)
 	if err != nil {
-		logger.Error(err, "failed to get controller statefulset")
+		logger.Error(err, "failed to get controller deployment")
 		return err
 	}
 
 	logger.Info("controller requires restart",
-		"ReadyReplicas", controllerStatefulset.Status.ReadyReplicas,
-		"Replicas", controllerStatefulset.Status.Replicas)
+		"ReadyReplicas", controllerDeployment.Status.ReadyReplicas,
+		"Replicas", controllerDeployment.Status.Replicas)
 	logger.Info("restarting csi controller")
 
-	err = r.getControllerPod(controllerStatefulset, controllerPod)
+	err = r.getControllerPod(controllerDeployment, controllerPod)
 	if errors.IsNotFound(err) {
 		return nil
 	} else if err != nil {
 		logger.Error(err, "failed to get controller pod")
 		return err
 	}
 
-	return r.restartControllerPodfromStatefulSet(logger, controllerStatefulset, controllerPod)
+	return r.restartControllerPodfromDeployment(logger, controllerDeployment, controllerPod)
 }
 
-func (r *CSIScaleOperatorReconciler) getControllerPod(controllerStatefulset *appsv1.StatefulSet, controllerPod *corev1.Pod) error {
-	controllerPodName := fmt.Sprintf("%s-0", controllerStatefulset.Name)
+func (r *CSIScaleOperatorReconciler) getControllerPod(controllerDeployment *appsv1.Deployment, controllerPod *corev1.Pod) error {
+	controllerPodName := fmt.Sprintf("%s-0", controllerDeployment.Name)
```

Comment:
This is operator pod right? if yes then operator pod won't have a suffix `<podname>-0`.

Reply:
Done, removed -0 from pod name.